src/core/Akka.Remote.Tests.Performance/InboundMessageDispatcherSpec.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Threading;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Util.Internal;
+using Google.ProtocolBuffers;
+using NBench;
+
+namespace Akka.Remote.Tests.Performance
+{
+    /// <summary>
+    /// Specs used to verify the performance of the <see cref="IInboundMessageDispatcher"/>,
+    /// one of the potential bottlenecks inside the remoting system
+    /// </summary>
+    public class InboundMessageDispatcherSpec
+    {
+        private const string MessageDispatcherThroughputCounterName = "InboundMessageDispatch";
+        private SerializedMessage _message;
+
+        private Counter _inboundMessageDispatcherCounter;
+        private static readonly AtomicCounter Counter = new AtomicCounter(0);
+
+        private ActorSystem _actorSystem;
+        private Address _systemAddress;
+        private IInboundMessageDispatcher _dispatcher;
+
+        private IInternalActorRef _targetActorRef;
+
+        /// <summary>
+        /// Not thread-safe, but called by a single thread in the benchmark
+        /// </summary>
+        private class BenchmarkActorRef : MinimalActorRef
```

Comment:
Using a `MinimalActorRef` implementation instead of an actor class so we can test this code in a single-threaded manner and not use any dispatchers.